docs/2.7.0/docs/daml/intro/12_Testing.rst

```diff
@@ -164,20 +164,35 @@ As you can see, the behavior of ``fetch``, ``fetchByKey`` and ``lookupByKey`` at
 Checking Coverage
 -----------------
 
-When ``daml test`` runs, it analyzes the ledger record to produce a report on what percentage of templates were created and which interface and template choices were exercised during our tests.
+When ``daml test`` runs a set of tests, it analyzes the ledger record from those tests to report template and choice coverage. It calculates what percentage of templates defined in the package were created, and what percentage of choices defined in the package were exercised.
 
-Flags Controlling Tests and Coverage
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+The user can also save the resulting coverage results for the test set to a file, and then read them back into a future report. In an invocation of ``daml test``, one can both read results in and run tests simultaneously, in order to generate a final report which aggregates them all. More details on the workflows this enables are detailed in `Serializing Results Workflows`_.
 
-``daml test`` has four important options which affect the coverage report: ``--test-pattern PATTERN``, ``--files FILE``, ``--show-coverage``, and ``--all``:
+.. figure:: images/12_Testing/coverage_workflow.png
+   :alt: Flow chart visually explaining sources and sinks for coverage results.
 
-Enabling ``--show-coverage`` lists by name any templates, choices, and interfaces which are not covered. By default, the report only reports percentage of coverage.
+Flags Controlling Test Set
+~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-The remaining three options affect which tests are run. The coverage report is in aggregate for all tests run - tests that don't run do not count towards coverage.
+The user can control the set of tests run by ``daml test`` using ``--test-pattern PATTERN``, ``--files FILE``, and ``--all``.
 
 * Passing ``--test-pattern <PATTERN>`` runs only the local tests which match ``PATTERN``.
 * Passing ``--files <FILE>`` runs only the tests found in ``FILE``.
-* Enabling ``--all`` runs tests in dependency modules as well. Note: it is not affected by ``test-pattern`` - all external tests are run, and ``test-pattern`` only restricts local tests.
+* Enabling ``--all`` runs tests in dependency modules as well. Note: all external tests are run, regardless of the setting in ``test-pattern`` - ``test-pattern`` only restricts local tests.
+
+Flags Controlling Serialization
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The user can save the final coverage results of a ``daml test`` invocation using ``--save-coverage FILE``. This will write the list of templates and choices in scope, along with the list of templates created and choices exercised.
```

Comment:
```suggestion
You can save the final coverage results of a ``daml test`` invocation using ``--save-coverage FILE``. This will write the list of templates and choices in scope, along with the list of templates created and choices exercised.
```